python/cugraph/tests/test_graph.py

```diff
@@ -793,6 +793,48 @@ def test_has_node(managed, pool, graph_file):
         assert G.has_node(n)
 
 
+# Test all combinations of default/managed and pooled/non-pooled allocation
+@pytest.mark.parametrize('managed, pool',
+                         list(product([False, True], [False, True])))
+@pytest.mark.parametrize('graph_file', DATASETS2)
+def test_bipartite_api(managed, pool, graph_file):
+    gc.collect()
+
+    rmm.reinitialize(
+        managed_memory=managed,
+        pool_allocator=pool,
+        initial_pool_size=2 << 27
+    )
+
+    assert(rmm.is_initialized())
+
+    cu_M = utils.read_csv_file(graph_file)
+    nodes = cudf.concat([cu_M['0'], cu_M['1']]).unique()
+
+    # Create set of nodes for partition
+    set1_exp = cudf.Series(nodes[0:int(len(nodes)/2)])
+    set2_exp = cudf.Series(set(nodes) - set(set1_exp))
+
+    G = cugraph.Graph()
+    assert not G.is_bipartite()
+
+    # Add a set of nodes present in one partition
+    G.add_nodes_from(set1_exp, bipartite='set1')
+    G.from_cudf_edgelist(cu_M, source='0', destination='1')
+
+    # Check if Graph is bipartite. It should return True since we have
+    # added the partition in add_nodes_from()
+    assert G.is_bipartite()
+
+    # Call sets() to get the bipartite set of nodes.
+    set1, set2 = G.sets()
+
+    # assert if the input set1_exp is same as returned bipartite set1
+    assert set1.equals(set1_exp)
+    # assert if set2 is the remaining set of nodes not in set1_exp
+    assert set2.equals(set2_exp)
```

Comment:
This tests the mechanics of the bipartite class.  I'm comfortable with the API based on this test code.

However, it might be worth noting that neither of the data sets in DATASETS2 (karate and dolphin) are actually bipartite.  This test, while mechanically correct, might be misleading.  Should we find/create a bipartite test data set?

Perhaps we should do the following:
 1. make a note in this test case that we're testing the mechanics of the API and that these data sets are not actually bipartite
 1. create an issue to add some bipartite test data to the test graphs and update this test case once that is available

Reply:
Made a note in the comments